src/service/inactive-administrative-cost-service.ts

```diff
@@ -0,0 +1,320 @@
+   */
+  public async checkInactiveUsers(params: InactiveAdministrativeCostFilterParameters)
+    : Promise<UserToInactiveAdministrativeCostResponse[]> {
+    const { notification } = params;
+    const differenceDate = notification ? 2 : 3;
+
+    const users = await User.find();
+    const eligibleUsers: UserToInactiveAdministrativeCostResponse[] = [];
+
+    // go through all users and get their last transfer and transaction
+    for (let i = 0; i < users.length; i += 1) {
+      const user = users[i];
+      if (!EligibleInactiveUsers.includes(user.type)) continue;
+      if (notification && user.inactiveNotificationSend) continue;
+
+      let isNotEligible = false;
+
+      // Find transfers and transaction of users and if not null reduce to the last one
+      const userTransfers = (await Transfer.find({
+        where: { fromId: user.id },
+        relations: { inactiveAdministrativeCost: true },
+      }));
+
+      const lastTransfer = userTransfers.length == 0 ? null : userTransfers
+        .reduce((prev, curr) => (((prev.createdAt < curr.createdAt) && curr.inactiveAdministrativeCost == null) ? curr : prev));
+      const userTransactions = ((await Transaction.find({ relations: ['from'] }))
+        .filter((t) => t.from.id === user.id));
+      const lastTransaction = userTransactions.length == 0  ? null : userTransactions
+        .reduce((prev, curr) => (prev.createdAt < curr.createdAt ? curr : prev));
+
+      if (lastTransfer != null) if (InactiveAdministrativeCostService.yearDifference(lastTransfer.createdAt) < differenceDate) {
+        isNotEligible = true;
+      }
+      if (lastTransaction != null) if (InactiveAdministrativeCostService.yearDifference(lastTransaction.createdAt) < differenceDate) {
+        isNotEligible = true;
+      }
+
+      if (!isNotEligible) {
+        const response: UserToInactiveAdministrativeCostResponse = { userId: user.id };
+        eligibleUsers.push(response);
+      }
+    }
+
+    return eligibleUsers;
+  }
+
+  /**
+   * Deletes the given InactiveAdministrativeCost and creates an undo transfer
+   * @param inactiveAdministrativeCostId
+   */
+  public async deleteInactiveAdministrativeCost(inactiveAdministrativeCostId: number)
+    : Promise<InactiveAdministrativeCost | undefined> {
+    // Find base inactive administrative cost entity.
+    const inactiveAdministrativeCost = await this.manager.findOne(InactiveAdministrativeCost, { ...InactiveAdministrativeCostService.getOptions({ inactiveAdministrativeCostId }) });
+    if (!inactiveAdministrativeCost) return undefined;
+
+    // Get amount from transfer
+    const amount: DineroObjectRequest = inactiveAdministrativeCost.transfer.amountInclVat.toObject();
+
+    // We create an undo transfer that sends the money back to the person.
+    const undoTransfer: TransferRequest = {
+      amount,
+      description: 'Deletion of InactiveAdministrativeCost',
+      fromId: 0,
+      toId: inactiveAdministrativeCost.fromId,
+    };
```

Comment:
research shows that it is a must that you link this to the inactive administration entity. I would say a `deleteTransfer` or whatever I called it in invoices.

Reply:
creditTransfer but I have changed this.